### DISCUSSION TITLE
Call to `/wave` is failing, locally

### DISCUSSION THREAD

**Discussion by edalzell** [qruto/laravel-wave#44] [Q&A]
Locally I see the call to `/wave` timeout, all the time, makes testing the chat feature I have very difficult, any ideas how to resolve?

![CleanShot 2023-11-13 at 12 01 01@2x](https://github.com/qruto/laravel-wave/assets/6069653/fed691a8-da51-41c3-8c33-d069b91eef2b)

I'm on  the latest version, running Herd.

Is there an nginx thing maybe?

**Comment by slavarazum** [comment]
Output buffering? 🤔😀

https://github.com/qruto/laravel-wave/issues/35#issuecomment-1689053174

**Reply by edalzell** [reply]
Nope I have that on, and not getting an error anywhere I can see.

**Reply by slavarazum** [reply]
Herd's nginx is not configurable. Seems like it has 10s timeout for inactive connections.
To preserve connection active, try to use [**ping** command](https://github.com/qruto/laravel-wave#manual-ping-control) with 8s interval:

```bash
php artisan sse:ping --interval=8
```

**Reply by edalzell** [reply]
it is configurable, I added the same lines you mention in the instructions, no change.

**Reply by slavarazum** [reply]
I missed that we can configure in the same way as Valet or change general config in **/Users/razum/Library/Application Support/Herd/config/nginx**

Have you tried the ping command?

**Reply by slavarazum** [reply]
It seems that the instructions in the readme are not suitable for Herd. Upon investigating Herd's nginx configs, I discovered that it has a default timeout of 60 seconds. Therefore, you can leave the configurations as they are by default, which will provide a stable connection of 60 seconds, usually sufficient for testing purposes.

If you need to extend the connection time, you can add `fastcgi_read_timeout 1h`, for instance, to the `location ~ [^/]\.php(/|$) {` section. Alternatively, you can run `php artisan sse:ping --interval=50` to maintain the connection.

I will rewrite docs for Herd setup soon.

**Comment by edalzell** [comment]
Seeing this in Firefox and Safari, locally, any ideas?
<img width="1139" alt="image" src="https://github.com/qruto/laravel-wave/assets/6069653/8a6564ad-4c08-4698-8048-63683165868f">

**Comment by slavarazum** [comment]
Configuration issue resolved.

**Reply by hotmeteor** [reply]
What was the problem / resolution? I have a team member hitting the same thing.

**Reply by edalzell** [reply]
I don't remember sorry it was a while back...
